contracts/SuiBridge.sol

```diff
@@ -126,11 +134,14 @@ contract SuiBridge is
             );
     }
 
+    // Same: pass the payload instead of the full message
     function _processTokenTransferMessage(bytes memory message) internal whenNotPaused {
         // Decode the message
+        // interesting, why? it does not look like a lot of data to decode
         // TODO: this is causing a "Stack Too Deep" error. Need to refactor
         // https://soliditydeveloper.com/stacktoodeep
         (
+            // Not all of the unit are unit256, please check the notion page
```

Comment:
again, haven't addressed this anywhere (focused on overall contract structure). having @georgedigkas tackle going through and altering types to align with the design in notion.